lib/puppet/ssl/host.rb

```diff
@@ -371,27 +378,61 @@ def state
 
   private
 
+  # Ensures that CRL is either on disk, or that CRL checking has been disabled
+  def ensure_crl!
+    if crl_usage && !Puppet::FileSystem.exist?(crl_path)
+      # The CertificateRevocationList indirector will attempt to download the
+      # CRL from the CA if it does not exist on disk.
+      if !CertificateRevocationList.indirection.find(name)
+        Puppet.debug _("Certificate revocation checking is enabled but a CRL cannot be found; CRL checking will not be performed.")
+        @crl_usage = false
+      end
+    end
+  end
+
+  # @param path [String] Path to CRL Chain
+  # @return [Array<OpenSSL::X509::CRL>] CRLs from chain
+  # @raise [Errno::ENOENT] if file does not exist
+  # @raise [Puppet::Error<OpenSSL::X509::CRLError>] if the CRL chain is malformed
+  def load_crls(path)
+    ending = "-----END X509 CRL-----\n"
```

Comment:
This is gross. I didn't see a parser in `OpenSSL` for loading potentially multiple PEMs (like `pemToObjects` in `jvm-ssl-utils`) but I maybe missed it.